src/plugins/interactive_setup/server/plugin.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import type { CorePreboot, PrebootPlugin, PluginInitializerContext } from 'src/core/server';
+import fs from 'fs/promises';
+import { errors } from '@elastic/elasticsearch';
+import Boom from '@hapi/boom';
+import chalk from 'chalk';
+import type { ConfigType } from './config';
+
+export class InteractiveSetupPlugin implements PrebootPlugin {
+  readonly #initializerContext: PluginInitializerContext<ConfigType>;
+  constructor(initializerContext: PluginInitializerContext<ConfigType>) {
+    this.#initializerContext = initializerContext;
+  }
+
+  public setup(core: CorePreboot) {
+    const { enabled } = this.#initializerContext.config.get<ConfigType>();
+
+    const isDev = this.#initializerContext.env.mode.dev;
+    const isConfigured =
+      core.elasticsearch.config.credentialsSpecified && core.elasticsearch.config.hosts.length;
+
+    /**
+     * env: dev -> enabled: false -> skip
+     * env: dev -> enabled: undefined -> skip
+     * env: dev -> enabled: true -> hold setup
+     * env: prod -> enabled: false -> skip
+     * env: prod -> enabled: undefined -> hold setup if not configured
+     * env: prod -> enabled: true -> hold setup if not configured
+     */
+    const shouldHoldSetupPhase = isDev
+      ? enabled === true
+      : enabled !== false
+      ? !isConfigured
+      : false;
```

Comment:
`yarn es snapshot` and `yarn start` automatically start local Elasticsearch instance and configure Kibana so we need to explore if there's an automated way of retrieving an enrolment token from Elasticsearch, or if developers have to manually go through interactive setup flow, or whether we disable TLS in development mode. 

Either way it will be good to force Kibana into interactive setup mode for local testing.

Reply:
So here is what we have currently:

```bash
# no TLS is used, no need in interactive mode
$ yarn es snapshot & yarn start

# TLS is properly and automatically configured, no need in interactive mode
$ yarn es snapshot --ssl & yarn start --ssl
```

Kibana developers basically don't need an interactive setup mode at all during development since our tooling allows them to easily setup TLS. But for us, to test this mode, we can probably just add a `--userSetup.forceSetup=true`?

Reply:
I agree, sounds good!